### PR TITLE
ASimAuditEventMicrosoftWindowsEvents.yaml-4

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for WindowsEvent,SecurityEvent and added WindowsEvent,SecurityEVENT parser in it. In BaracudaWAF file removed WindowsEvent parser and in ASimAuditEvent added WindowsEvent Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the WindowsEvent andEvent parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check for below failed validation. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please investigate the validation failures. Thanks

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+                    // SecurityEvents
+                    SecurityEvent
+                    | where not(disabled)
+                    | where (isnull(starttime) or TimeGenerated >= starttime) 
+                        and (isnull(endtime) or TimeGenerated <= endtime)
+                        and EventID in(FilteredEventIds)
+                    | where (array_length(srcipaddr_has_any_prefix) == 0)
+                        and (array_length(actorusername_has_any) == 0 or EventData has_any (actorusername_has_any))
+                        and (array_length(newvalue_has_any) == 0 or EventData has_any (newvalue_has_any))         
+                    | project EventID, EventData, _ResourceId, TimeGenerated, Computer, Type, _ItemId
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml:132]: done automated testing. Following are the results, can you please check?
FAIL: tests_main_func (__main__.FilteringTest.tests_main_func) (<subtest>)
----------------------------------------------------------------------
AssertionError: 0 == 0 : Parameter: actorusername_has_any - Got no results at all after filtering. Filtered by value: @'SECCXP\billb'

======================================================================
FAIL: tests_main_func (__main__.FilteringTest.tests_main_func) (<subtest>)
----------------------------------------------------------------------
AssertionError: 0 == 0 : Parameter: actorusername_has_any - Got no results at all after filtering. Filtered by value: @'SECCXP\billb',@'NT AUTHORITY\SYSTEM'

======================================================================
FAIL: tests_main_func (__main__.FilteringTest.tests_main_func) (<subtest>)
----------------------------------------------------------------------
AssertionError: 1 == 1 : Only one value exists for parameter: eventresult - validations for this parameter are partial

----------------------------------------------------------------------
Ran 1 test in 142.026s

FAILED (failures=3)
::error::Tests failed for Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftEvent.yaml [hunk 2]
```diff
@@ -0,0 +1,259 @@
+Parser:
+    Title: Audit Event ASIM filtering parser for Microsoft Windows Events audit events
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftEvent.yaml:2]: couldn't find data. did we do data and filter testing using ASC for this parser?

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftWindowsEvents.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftWindowsEvents.yaml:1]: couldn't find data. Did we do data and filtering testing for this parser using ASC?

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventMicrosoftEvent.yaml [hunk 4]
```diff
@@ -0,0 +1,184 @@
+Parser:
+    Title: Audit Event ASIM parser for Microsoft Windows Events audit events
+    Version: "0.2.1"
+    LastUpdated: June 7, 2024
+Product:
+    Name: Microsoft Windows
+Normalization:
+    Schema: AuditEvent
+    Version: "0.1.0"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventMicrosoftEvent.yaml:9]: done

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventMicrosoftSecurityEvents.yaml [hunk 5]
```diff
@@ -0,0 +1,201 @@
+Parser:
+    Title: Audit Event ASIM parser for Microsoft Windows Events audit events
+    Version: "0.2.1"
+    LastUpdated: June 7, 2024
+Product:
+    Name: Microsoft Windows
+Normalization:
+    Schema: AuditEvent
+    Version: "0.1.0"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventMicrosoftSecurityEvents.yaml:9]: done

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventMicrosoftWindowsEvents.yaml [hunk 6]
```diff
@@ -15,7 +15,7 @@ References:
     - Title: Office 365 Management Activity API schema
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventMicrosoftWindowsEvents.yaml:15]: done

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftWindowsEvents.yaml (hunk 3 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftWindowsEvents.yaml:1]: yes

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
One suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button. Thanks

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftEvent.yaml [hunk 7]
```diff
@@ -0,0 +1,259 @@
+Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftEvent.yaml:1]: done

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml (hunk 1 above)
Comment (vakohl) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml:132]: can't see the picture

Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml:132]: testing done ASC portal

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**